miner/block_builder.go

```diff
@@ -280,14 +287,18 @@ func (t *BlockBuilder) createBlockLoop(ctx context.Context) {
 				continue
 			}
 
+			started := time.Now()
+
 			atxID, proofs, atxs, err := t.blockOracle.BlockEligible(layerID)
 			if err != nil {
 				events.ReportDoneCreatingBlock(true, layerID.Uint32(), "failed to check for block eligibility")
+				t.saveBlockBuildDurationMetric(ctx, started, layerID, types.BlockID{})
```

Comment:
i think we should use this metric only for a successful build. maybe we can time failed builds as well separately, but i don't see why it would be valuable

Reply:
Do we need to split the metric into two: with and without network broadcast time?

Reply:
we just need to collect it before sending to the network